### DISCUSSION TITLE
Non-empty .wav file is rendered but unplayable

### DISCUSSION THREAD

**Discussion by samuelbraun04** [DBraun/DawDreamer#151] [Q&A]
I am trying to load a preset onto the [Serenity SE](https://www.adsrsounds.com/product/software/quiet-music-serenity-se-relaxing-pads-vst-sampler/) plugin, load a midi file, and render it into a wave file.

The code I'm doing this with is:

```
ENGINE.set_bpm(30)
serenity = ENGINE.make_plugin_processor("serenity", path/to/plugin)
serenity.load_vst3_preset(path/to/preset/example.vstpreset)
serenity.load_midi(path/to/midi, beats=True)
        
graph = [
    (serenity, [])
]
        
ENGINE.load_graph(graph)
ENGINE.render(4, beats=True)
wavfile.write(path/to/finafile, 44100, (ENGINE.get_audio()).transpose())
```

When I run this [a non-empty wav file](https://drive.google.com/drive/folders/1eJ3aAkwPKLCTByeZAiCCDWiN4nSzR1pu?usp=sharing) is rendered, but it is unplayable (I've tried on multiple different players, and when I drag-and-drop it into FL Studio the clip appears empty).

Something like this has worked for another plugin (specifically [Vital](https://vital.audio/) and I was loading states), so what could be the problem here? Thank you.

**Comment by DBraun** [comment]
The rendered wav file is 32 channels of zeros. I recommend Audacity for inspecting waveforms. On Windows I also do control-shift-f to view all the tracks at once.

I suspect that the loading of the preset file isn't working. Have you tried `serenity.open_editor()` in order to select a preset?

**Reply by samuelbraun04** [reply]
Thanks for the audacity tip I didn't know about that.

When I do serenity.open_editor() and select a preset I get the same rendered file.

![image](https://user-images.githubusercontent.com/90664891/222578365-5180cb64-008d-4752-a5b0-4c7b59f4c4fb.png)


The code I'm running:

```
ENGINE.set_bpm(30)
serenity = ENGINE.make_plugin_processor("serenity", path/to/plugin)
#serenity.load_vst3_preset(path/to/preset/example.vstpreset)
serenity.open_editor()
serenity.load_midi(path/to/midi, beats=True)
        
graph = [
    (serenity, [])
]
        
ENGINE.load_graph(graph)
ENGINE.render(4, beats=True)
wavfile.write(path/to/finafile, 44100, (ENGINE.get_audio()).transpose())
```

**Reply by samuelbraun04** [reply]
The problem is with the plugin itself. Other plugins work, and just using the Serenity plugin alone with no presets doesn't work either:
```
ENGINE.set_bpm(30)
serenity = ENGINE.make_plugin_processor("serenity", path/to/plugin)
serenity.load_midi(path/to/midi, beats=True)
        
graph = [
    (serenity, [])
]
        
ENGINE.load_graph(graph)
ENGINE.render(4, beats=True)
wavfile.write(path/to/finafile, 44100, (ENGINE.get_audio()).transpose())
```

**Reply by DBraun** [reply]
Since I don't have the plugin, can you try this
```python
assert serenity.can_set_bus(0, 2)
serenity.set_bus(0, 2)
```
And if that doesn't work
```python
assert serenity.can_set_bus(2, 2)
serenity.set_bus(2, 2)
```
And a low probability that this could help, but try just the others first
```python
serenity.disable_nonmain_buses()
```

Also, make sure the MIDI notes make sense for the plugin. Maybe
```python
serenity.add_midi_note(60, 127, 0.1, 8)
```
could be more reliable.

**Reply by samuelbraun04** [reply]
Unfortunately, ```serenity.set_bus(0, 2)``` returns ```serenity CANNOT ApplyBusesLayout inputs: 0 outputs: 2```
and ```serenity.set_bus(2, 2)``` returns ```serenity CANNOT ApplyBusesLayout inputs: 2 outputs: 2```

```serenity.disable_nonmain_buses()``` returns the error ```Exception has occurred: AttributeError
'dawdreamer.dawdreamer.PluginProcessor' object has no attribute 'disable_nonmain_buses'```

And solely doing ```serenity.add_midi_note(60, 127, 0.1, 8)``` still returns an empty wav file.

There is a [free version of the plugin available](https://quietmusic.eu/serenity), but I might just have to find an alternative since these problems seem to be something weird about the plugin itself.
